var/spack/repos/builtin/packages/py-dask-ml/package.py

```diff
@@ -0,0 +1,86 @@
+    depends_on('py-pandas@0.24.2:', type=('build', 'run'))
+    depends_on('py-scikit-learn@0.23:', type=('build', 'run'))
+    depends_on('py-scipy', type=('build', 'run'))
+    depends_on('py-dask-glm@0.2.0:', type=('build', 'run'))
+    depends_on('py-multipledispatch@0.4.9:', type=('build', 'run'))
+    depends_on('py-packaging', type=('build', 'run'))
+
+    depends_on('py-black', type=('build', 'run'), when='+docs')
+    depends_on('py-coverage', type=('build', 'run'), when='+docs')
+    depends_on('py-flake8', type=('build', 'run'), when='+docs')
+    depends_on('py-graphviz', type=('build', 'run'), when='+docs')
+    depends_on('py-heapdict', type=('build', 'run'), when='+docs')
+    depends_on('py-ipykernel', type=('build', 'run'), when='+docs')
+    depends_on('py-ipython', type=('build', 'run'), when='+docs')
+    depends_on('py-isort', type=('build', 'run'), when='+docs')
+    depends_on('py-mypy', type=('build', 'run'), when='+docs')
+    depends_on('py-nbsphinx', type=('build', 'run'), when='+docs')
+    depends_on('py-nose', type=('build', 'run'), when='+docs')
+    depends_on('py-numpydoc', type=('build', 'run'), when='+docs')
+    depends_on('py-psutil', type=('build', 'run'), when='+docs')
+    depends_on('py-pytest', type=('build', 'run'), when='+docs')
+    depends_on('py-pytest-cov', type=('build', 'run'), when='+docs')
+    depends_on('py-pytest-mock', type=('build', 'run'), when='+docs')
+    depends_on('py-sortedcontainers', type=('build', 'run'), when='+docs')
+    depends_on('py-sparse', type=('build', 'run'), when='+docs')
+    depends_on('py-sphinx', type=('build', 'run'), when='+docs')
+    depends_on('py-sphinx-rtd-theme', type=('build', 'run'), when='+docs')
+    depends_on('py-sphinx-gallery', type=('build', 'run'), when='+docs')
+    depends_on('py-testpath', type=('build', 'run'), when='+docs')
+    depends_on('py-tornado', type=('build', 'run'), when='+docs')
+    depends_on('py-toolz', type=('build', 'run'), when='+docs')
+    depends_on('py-zict', type=('build', 'run'), when='+docs')
+    depends_on('py-dask-sphinx-theme@1.1.0:', type=('build', 'run'), when='+docs')
+    depends_on('py-nbsphinx', type=('build', 'run'), when='+docs')
+
+    depends_on('py-xgboost+dask', type=('build', 'run'), when='+docs')
+    depends_on('py-xgboost+dask', type=('build', 'run'), when='+xgboost')
+
+    patch('xgboost_dependency.patch')
+
+    conflicts('+docs', when='%gcc target=aarch64:')
+
+    def install(self, spec, prefix):
+        pip = which('pip')
+
+        if '+xgboost' in spec:
+            pip('install', self.stage.source_path, '.[xgboost]',
+                '--prefix={0}'.format(prefix))
+        else:
+            pip('install', self.stage.source_path, '--prefix={0}'.format(prefix))
+
+        if '+docs' in spec:
+            with working_dir('docs'):
+                make('html')
+                install_tree(
+                    join_path(self.stage.source_path, 'docs'), self.prefix.docs)
```

Comment:
I would do something like:
```python
@run_after('build')
def build_docs(self):
    if '+docs' in self.spec:
        with working_dir('docs'):
            make('html')

@run_after('install')
def install_docs(self):
    if '+docs' in self.spec:
        install_tree('docs', self.prefix.docs)
```

Reply:
I used suggested functions.

Reply:
I'm having problems not finding the `dask-ml` module similar to the following issue, so I'll change the docs build after install.
https://github.com/dask/dask-ml/issues/693